[Chiang-HuiXin] iP

Chiang-HuiXin opened the pull request.

Chiang-HuiXin committed:
Added Eva.java file for level 0

I made a new file called Eva.java for my chatbot named Eva and coded it to greet and exit

Chiang-HuiXin committed:
Update code according to the requirements of level 1: Echo

Add lines in between the texts to increase readibility. Add while loop to echo messages.

Chiang-HuiXin committed:
Add code for level 2

Created an array of size 100 for list.
Add String into array when scanned.
Iterate the list and print out in bullet points when user type list.

Chiang-HuiXin committed:
Add code for level 3

Created an array of size 100 for list.
Add String into array when scanned.
Iterate the list and print out in bullet points when user type list.

Chiang-HuiXin committed:
Add Task.java and update code according to requirements of level 3

I created class Task as recommeded in the question.

For the code in Eva.java, I added a few more if branches to account for the mark and unmark feature.
I created helper functions to increase the readibility of the main code.

Chiang-HuiXin committed:
Update code to tweak the code to follow coding standards

I changed markAsDone() function name to setMarkAsDone() as it is a setter function.

TrungBui32 reviewed (commented):
Overall, good cood!

anlamm commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
name of boolean is good

anlamm commented on src/main/java/Eva.java:
```diff
@@ -0,0 +1,73 @@
+        int count = 0;
+        Task[] sentenceArray = new Task[100];
+
+        while(true)
```
should put the "{" in line 51

anlamm commented on src/main/java/Eva.java:
```diff
@@ -0,0 +1,73 @@
+            line = in.nextLine();
+            sentenceArray[count] = new Task(line);
+
+            if(line.equalsIgnoreCase("list")) {
```
if..else if ... else layout looks great

ChuaCleon commented on src/main/java/Eva.java:
```diff
@@ -0,0 +1,73 @@
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    public static void markTaskAsDone(Task[] tasks, String line) {
```
would markTask be a better function name? conveys the same meaning but is slightly shorter and improves readability. Same goes with unmark function below

ChuaCleon commented on src/main/java/Eva.java:
```diff
@@ -0,0 +1,73 @@
+
+public class Eva {
+
+    public static final String HORIZONTAL_LINE = "---------------------------------------------------------------";
```
good job on initialising the horizontal line as a constant, with good naming conventions

Chiang-HuiXin committed:
Add Deadline.java class file

Class Deadline is an inheritance of Task.
Deadline has an additional by variable which is used to keep track of the duedates.

Chiang-HuiXin committed:
Add Todo.java

Todo is an inheritance of Task.
There is a toString() function that overrides to display the additional [T].

Chiang-HuiXin committed:
Add Event.java

Event is an inheritance of Task.
It contains 2 additional variables 'from' and 'to' to keep track of the time period of the event.

Chiang-HuiXin committed:
Update Eva.java

A new toString() function is added to print the [ ] out.
With this function, its subclass can override this to print their own version.

Chiang-HuiXin committed:
Update code according to requirements of level 4

I changes the if-else branches to switch branch to increase readability.
Added new cases for todo, deadline, event and their respective functions.
Changed the names from markTaskAsDone to markTask to increase readability as it is shorter.

Chiang-HuiXin committed:
Add TaskManager.java

Moved task handling logic (add, mark, unmark, list) from the main loop to this new TaskManager class.
This class is also incharge of the printing of tasks, todos, deadlines and events.
This change simplifies the code in main.

Chiang-HuiXin committed:
Update main code to simplify it

Created a new class called TaskManager and shift all the task handling logic to the new class.
This shortens and simplifies the main method, increasing its readability.
Also removed magic numbers.

Chiang-HuiXin committed:
Add EvaException.java to handle exceptions

Chiang-HuiXin committed:
Update Eva.java to handle exceptions

Updated to handle the situation where the user type in an unknown
command (not todo, deadline, event, mark etc)

Previously, if user type something, it will return added: (something)
Removed it as this feature (level 2) is no longer relevant and it
does not make sense with the exception mentioned above.

Chiang-HuiXin committed:
Update code to handle other exceptions

The errors that are being handled are:
*missing descriptions in todo, deadline and event
*missing by in deadline
*missing from and to in event

When the exceptions are called, a message will be shown to guide
users to enter in the correct format

Chiang-HuiXin committed:
Update exception handling for markTask and unmarkTask

It handles case where the task number is out of range and prompts
users to input a valid task number if an error occurs

Chiang-HuiXin committed:
Merge branch 'branch-Level-5'

Chiang-HuiXin committed:
Change tasks from array to ArrayList

Chiang-HuiXin committed:
Update code for delete feature

- Implemented deleteTask method to allow users to delete tasks by task
number
- Ensured that task number validation is performed, with appropriate
error handling using EvaException.
- After deletion, the task count is updated and a confirmation message
is printed along with the remaining number of tasks.

Chiang-HuiXin committed:
Add saveTasksToFile method for file saving

-Creates "data" directory if it does not exists
-Add exception handling for potential IO errors

Chiang-HuiXin committed:
Add loadTasksFromFile method for file saving

Implement method to load tasks from file

Chiang-HuiXin committed:
Add parseTaskFromString method for file saving

Add task parsing logic to convert string input into objects based
on task type

Chiang-HuiXin committed:
Update code for level 6 and level 7

Allows for file saving and deletion of tasks

Chiang-HuiXin committed:
Merge branch 'branch-Level-6'

Chiang-HuiXin committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/TaskManager.java

okkhoy commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,237 @@
+import java.util.ArrayList;
+import java.io.*;
```
don't use wildcard imports.

okkhoy commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,237 @@
+        String[] parts = line.replaceFirst("deadline", "").split("/by");
+
+        if (parts.length < 2) {
+            throw new EvaException("Oh no! The deadline command must have a description and a by time." +
```
for easy maintenance, better to declare a variable and use the variable in the exception throwing

okkhoy commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,237 @@
+
+    public void printEvent(String line) throws EvaException {
+
+        String[] eventParts = line.replaceFirst("event", "").split("/from|/to");
```
can SLAP better here

okkhoy commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,237 @@
+            File directory = new File("data");
+            if (!directory.exists()) {
+                boolean dirCreated = directory.mkdirs();
+                if (!dirCreated) {
+                    System.out.println("An error occurred: Could not create the directory.");
+                    return;
```
try using exception here

okkhoy reviewed (commented):
You have not packaged the code as per A-Packages. Please do so.

Chiang-HuiXin committed:
Implement Storage class file

Moved the file save management from TaskManager to Storage

Chiang-HuiXin committed:
Update TaskManager to implement Storage class

Chiang-HuiXin committed:
Update Storage to remove a bug

Added trim() functions.
Previously, when recovering the saved data and printing it out, there
will be extra spaces. For example, originally it was 1. [T][ ] code, it
will become 1. [T][ ]  code, with an extra space.

Chiang-HuiXin committed:
Implement Ui class

Responsible with the interactions with the user.

Chiang-HuiXin committed:
Update Eva.java with Ui class

Allows the code to be more concise and readable

Chiang-HuiXin committed:
Update TaskManager.java with Ui class

- Changed all the System.out.println with ui.showMessage.
- This allows Ui class to only handle the interactions between the
user

Chiang-HuiXin committed:
Implement CommandHandler and Parser class file

CommandHandler is responsible for executing commands by invoking
relevant methods from TaskManager

Parser is responsible for parsing the command and extracting arguments
from the input string

Chiang-HuiXin committed:
Simplify code in Eva.java

Eva.java is simplified to only handling the starting of program and
ending of program.
This makes the main code shorter and improves readability

Chiang-HuiXin committed:
Update to reduce the responsibilities of TaskManager

Previously, TaskManager was parsing and doing complex logics like
todos, events and deadlines.
Now, with Parser class, the parsing responsibles are moved to Parser
class allowing TaskManager to only handle task related responsibilities

Chiang-HuiXin committed:
Update CommandHandler to include case "find"

Chiang-HuiXin committed:
Update Parser to parse find inputs

Chiang-HuiXin committed:
Add contains method in Task

contains will check if a description contains a word

Chiang-HuiXin committed:
Update TaskManager to add find operator

- findTasks will find if a specific word is in the list
- If there is, it will print out the tasks with that word
- If not, it will print a message out

Chiang-HuiXin committed:
Change contains to isContain

To make the boolean method sound like a boolean

Chiang-HuiXin committed:
Update TaskManager to remove an unnecessary method

- Removed extractDigit because the parsing logic is already done in
Parser class
- As parseArguments in Parser class return String[], it is required
to use Interger.parseInt to convert from String to int

Chiang-HuiXin committed:
Update TaskManager to remove an unnecessary method

- Removed extractDigit because the parsing logic is already done in
Parser class
- As parseArguments in Parser class return String[], it is required
to use Interger.parseInt to convert from String to int

Chiang-HuiXin committed:
Add JavaDoc to the classes and methods

Chiang-HuiXin committed:
Add JavaDoc to the classes and methods

Chiang-HuiXin committed:
Change the names of some methods

Decided to change printTodo, printDeadline, printEvent to addTodo,
addDeadline, addEvent because the methods is about adding those task
into tasklist. So print is not an accurate representation of the
method.

Chiang-HuiXin committed:
Change the names of some methods

Decided to change printTodo, printDeadline, printEvent to addTodo,
addDeadline, addEvent because the methods is about adding those task
into tasklist. So print is not an accurate representation of the
method.

Chiang-HuiXin committed:
Change the names of some methods

Decided to change printTodo, printDeadline, printEvent to addTodo,
addDeadline, addEvent because the methods is about adding those task
into tasklist. So print is not an accurate representation of the
method.

Chiang-HuiXin committed:
Change the names of some methods

Decided to change printTodo, printDeadline, printEvent to addTodo,
addDeadline, addEvent because the methods is about adding those task
into tasklist. So print is not an accurate representation of the
method.

Chiang-HuiXin committed:
Merge pull request #1 from Chiang-HuiXin/branch-Level-9

Add find feature to code

Chiang-HuiXin committed:
Revert "Change the names of some methods"

This reverts commit 98171bfb5c160c886fc87a89a5af847e04b4de5b.

Chiang-HuiXin committed:
Revert "Change the names of some methods"

This reverts commit 75d0dc422a49b96fc657f598e2def5aa338737f6.

Chiang-HuiXin committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc-old' into branch-A-JavaDoc

Chiang-HuiXin committed:
Merge pull request #3 from Chiang-HuiXin/branch-A-JavaDoc

Add JavaDoc for documentation

Chiang-HuiXin committed:
Add user guide to README.md

Chiang-HuiXin committed:
Add new methods to Ui.java

These methods are responsible in dealing the Ui with users.
Initially, this is in the methods in TaskManager which is
inappropriate as methods in TaskManager show deal with logics
regarding commands/tasks.

Chiang-HuiXin committed:
Update code for TaskManager.java

- Simplify TaskManger to handle task management only
- This also increases readability

Chiang-HuiXin committed:
Update exception code to make it concise

- Notice the same exception code repeating in different methods
- Extract it out and create new method in EvaException
- Makes code more readable

Chiang-HuiXin committed:
Add javadoc for new methods in Ui.java and EvaException.java

Chiang-HuiXin committed:
Update Storage.java to ensure tasks stay marked

Initially, when tasks are marked. After exiting the program and
coming back, the marked tasks would not stay marked

Chiang-HuiXin committed:
Add javadoc in Storage.java

Chiang-HuiXin committed:
Clean up code: remove unused lines and redundant methods

Chiang-HuiXin committed:
Add packages to code

Chiang-HuiXin committed:
Delete original Duke.java

Chiang-HuiXin committed:
Refactor code to increase readability

- Shorten long methods such as parseTaskFromString in Storage file and
parseArguments in Parser to increase readability
- Add and update Javadoc for new extracted methods

Chiang-HuiXin committed:
Update Javadoc to follow coding conventions

Chiang-HuiXin committed:
Update Javadoc for some methods to adhere to coding standards

Chiang-HuiXin committed:
Amend README.md to correct a line
